packages/validator/src/services/attestation.ts

```diff
@@ -88,21 +98,37 @@ export class AttestationService {
     slot: Slot,
     committeeIndex: CommitteeIndex,
     duties: AttDutyAndProof[],
+    attestationNoCommittee: phase0.AttestationData,
     signal: AbortSignal
   ): Promise<void> {
-    // Step 1. Download, sign and publish an `Attestation` for each validator.
-    const attestation = await this.produceAndPublishAttestations(slot, committeeIndex, duties);
+    // Step 1. Mutate, sign and publish `Attestation` for each validator.
+    const attestation = await this.publishAttestations(slot, committeeIndex, attestationNoCommittee, duties);
 
     // Step 2. If an attestation was produced, make an aggregate.
     // First, wait until the `aggregation_production_instant` (2/3rds of the way though the slot)
-    await sleep(this.clock.msToSlotFraction(slot, 2 / 3), signal);
+    await sleep(this.clock.msToSlot(slot + 2 / 3), signal);
+    this.metrics?.attesterStepCallProduceAggregate.observe(this.clock.secFromSlot(slot));
 
     // Then download, sign and publish a `SignedAggregateAndProof` for each
     // validator that is elected to aggregate for this `slot` and
     // `committeeIndex`.
     await this.produceAndPublishAggregates(attestation, duties);
   }
 
+  /**
+   * Performs the first step of the attesting process: downloading one `Attestation` object.
+   * Beacon node's endpoint produceAttestationData return data is not dependant on committeeIndex.
+   * For a validator client with many validators this allows to do a single call for all committees
+   * in a slot, saving resources in both the vc and beacon node
```

Comment:
this is nice!

Reply:
Now in its own PR https://github.com/ChainSafe/lodestar/pull/3918